lib/Interpreter/CppInterOp.cpp

```diff
@@ -3267,4 +3274,181 @@ namespace Cpp {
                          complete_column);
   }
 
+#define DEBUG_TYPE "autoload"
+
+  static inline std::string DemangleNameForDlsym(const std::string& name) {
+    std::string nameForDlsym = name;
+
+#if defined(R__MACOSX) || defined(R__WIN32)
```

Comment:
Generally we avoid that to enable cross-compilation. We can ask the target triple for the same information at runtime. Should be something like `Interpreter::getExecutionEngine::getTargetTriple::getOS`.